src/components/backgroundselector/BackgroundService.js

```diff
@@ -214,20 +218,15 @@ goog.require('ga_glstylestorage_service');
 
         this.setById = function(map, newBgId) {
           if (map && (!bg || newBgId !== bg.id)) {
-            var newBg = getBgById(newBgId);
-            if (newBg) {
-              bg = newBg;
+            bg = getBgById(newBgId);
+            if (bg) {
```

Comment:
This can lead to errors potentially because you could assign ` undefined`   to the current bg (value returned by `gaBg.get` )  used through all the appliction. That's why this asignment was done after the if. I would leave it as it was.

Reply:
ok but we should not pass in undefined here. We should check that before calling setById. I don't like methods that silently fail, I don't think it's good design.

Reply:
I will leave though